computations/src/lib.rs

```diff
@@ -96,91 +110,104 @@ impl Accumulator {
     }
 
     /// Convert an arbitrary symbol to a computation tree
-    fn to_tree<'a>(s: &Sym) -> Result<ComputationTree, &'static str> {
-        match s {
+    fn to_tree<'a>(s: Positioned<&Sym>) -> Result<ComputationTree, Error> {
+        match s.value {
             Sym::CombS | Sym::CombK | Sym::CombD | Sym::CombI => {
-                Ok(ComputationTree::CombOp(Self::to_comb(s)))
+                Ok(ComputationTree::CombOp(Self::to_comb(s.value)))
             }
             Sym::Map | Sym::Eq | Sym::Add | Sym::And | Sym::Or | Sym::Filter | Sym::Reduce => {
-                Ok(ComputationTree::BinOpSym(Self::to_binary(s)))
+                Ok(ComputationTree::BinOpSym(Self::to_binary(s.value)))
+            }
+            Sym::Iota | Sym::Len | Sym::Neg => {
+                Ok(ComputationTree::UnOpSym(Self::to_unary(s.value)))
             }
-            Sym::Iota | Sym::Len | Sym::Neg => Ok(ComputationTree::UnOpSym(Self::to_unary(s))),
             Sym::Literal(n) => Ok(ComputationTree::Lit(n.clone().into())),
             Sym::Var(v) => Ok(ComputationTree::Lit(Literal::Var(v.clone()))),
-            Sym::Lambda(prog) => non_linear_check(prog).and_then(|body| {
-                Ok(ComputationTree::Lambda {
-                    vars: Self::count_variables(prog),
-                    body: Box::new(body),
-                })
+            Sym::Lambda(prog) => non_linear_check(
+                prog.iter().map(|s| &s.value).positioned(
```

Comment:
I think we should avoid such "on the fly" computation of default positions and provide another representation of programs that includes position information instead @SolarLiner

Reply:
You mean baking positions into `ComputationTree`? I don't think it is necessary to duplicate data structures here; otherwise yes, keeping positions around would be best.

Reply:
Yup. In that case let's just add a field for the positions in the ComputationTree

Reply:
We can use the `Spanned` wrapper type that has all the goodies to make manipulating positions and values easier within it